[SPIR-V] fix return type for OpAtomicCompareExchange

simeonschaub opened the pull request:
fixes #152863

Tests were written with some help from Copilot

simeonschaub committed:
[SPIR-V] fix return type for OpAtomicCompareExchange

fixes #152863

Tests were written with some help from Copilot

Naghasan commented on llvm/test/CodeGen/SPIRV/AtomicCompareExchange64BitPhiNode.ll:
```diff
@@ -0,0 +1,35 @@
+; RUN: llc -O0 -mtriple=spirv64-unknown-unknown %s -o - | FileCheck %s --check-prefix=CHECK-SPIRV
+; RUN: llc -O0 -mtriple=spirv64-unknown-unknown %s -o %t.spv -filetype=obj
+; RUN: spirv-val %t.spv
```
```suggestion
; RUN: %if spirv-tools %{ llc -O0 -mtriple=spirv64-unknown-unknown %s -o - -filetype=obj | spirv-val %}
```

spirv-val maybe not be available

simeonschaub committed:
relax test

I was testing based off of LLVM 20, looks like depending on the version
`OpConstant %8 0` might be optimized to `OpConstantNull %8`

simeonschaub committed:
don't assume spirv-val is available

Co-authored-by: Victor Lomuller <victor@codeplay.com>

simeonschaub committed:
simplify test

remove checks that fail on CI and are not directly related to the issue
